### ISSUE TITLE
`eas channel:list` fails in Github Action

### ISSUE THREAD

**Issue by zaniluca** [expo/eas-cli#1752]
### Summary

When trying to fetch the channels list from within a GitHub action script the command fails with exit code 1 without any output.

Even by adding `EXPO_DEBUG` nothing gets printed out

The exact same command works fine locally with `npx eas-cli` or a locally installed version

### Managed or bare?

Managed

### Environment

 expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 4.4 Ubuntu 20.04.1 LTS (Focal Fossa)
      Shell: 5.8 - /usr/bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
    npmPackages:
      expo: ^48.0.0 => 46.0.8
      react: 18.2.0 => 18.0.0
      react-dom: 18.2.0 => 18.0.0
      react-native: 0.71.3 => 0.69.6
      react-native-web: ~0.18.10 => 0.18.7
    npmGlobalPackages:
      eas-cli: 3.8.1
    Expo Workflow: managed

---

Expected package @expo/config-plugins@^5.0.0
Found invalid:
  @expo/config-plugins@4.0.18
  (for more info, run: npm why @expo/config-plugins)
Some dependencies are incompatible with the installed expo package version:
 - @shopify/flash-list - expected version: 1.1.0 - actual version installed: 1.4.0
 - expo-linking - expected version: ~3.2.4 - actual version installed: 3.2.2
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo doctor --fix-dependencies,
or install individual packages by running expo install [package-name ...]

### Error output

```
Run EXPO_DEBUG=true eas channel:list --json --non-interactive > channels.json
  
    Error: channel:list command failed.
Error: Process completed with exit code 1.
```

### Reproducible demo or steps to reproduce from a blank project

You can view the complete output of the script [here](https://github.com/zaniluca/ping-4-gitlab/actions/runs/4472383887/jobs/7858478944).

The whole script is the following but the command fails immediately so anything below the first line shouldn't matter
```bash
eas channel:list --json --non-interactive > channels.json
local channels=`jq '.currentPage[] .name' ./channels.json`

echo $channels
echo "Looking for $GITHUB_REF_NAME"

if printf '%s\0' "${channels[@]}" | grep -Fxqz -- '$GITHUB_REF_NAME'; then
  eas channel:delete --non-interactive $GITHUB_REF_NAME
  eas branch:delete --non-interactive $GITHUB_REF_NAME
else
  echo "Preview channel not found"
fi
```

**Comment by byCedric** [comment]
Hi @zaniluca! I just tried both the `eas channel:list --json --non-interactive` and your script, and it seems that I get a more helpful error ([see run](https://github.com/byCedric/expo-monorepo-example/actions/runs/4667322932/jobs/8262980477?pr=96#step:7:21)). Since `eas channel:list --json --non-interactive` worked fine, and the error pointing to the `local` syntax in your bash script, could this be related to your own bash script inside github actions?

Hope this helps!

**Comment by zaniluca** [comment]
@byCedric hi! thank you so much for your time!
I examined your run and workflow file and trying to replicate it I discovered what the error was.

Except for the bash syntax error (which for some reason I was not getting reported) I found out that the problem was I was not running `yarn install`, so the dynamic `app.config.ts` file complained about missing dependencies (dotenv in my case). That file includes the `extra.eas.projectId` that from what I understand is needed to reference the correct eas project. Because of that the `eas channel:list` was failing

Thanks again 🙏
